pkg/maintenance/postgresql.go

```diff
@@ -168,9 +168,12 @@ func (p *PostgreSQL) waitForUpgrade(ctx context.Context) error {
 			switch event.Type {
 			case watch.Modified:
 				ops, _ := event.Object.(*stackgresv1.SGDbOps)
+				if ops.Status.Conditions == nil || ops.Spec.Op != string(op) {
```

Comment:
I think there's no status whatsoever for non recycled CRD. Therefore you might want to check `status` for nil?

Reply:
Yeah but the status isn't a pointer :)

Reply:
Then no problem :D